src/components/Votes.jsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+import { patchArticle } from "../../Utils/api";
+
+const Votes = ({ newArticle }) => {
+  const originalVote = newArticle.votes;
+  const [newVotes, setNewVotes] = useState(originalVote);
+  const [voteChange, setVoteChange] = useState(0);
+  const [err, setErr] = useState(null);
+
+  const updateVotes = (changeValue) => {
+    if (voteChange === 0) {
+      patchArticle(newArticle.article_id, changeValue)
+        .then(() => {
+          setNewVotes(newVotes + changeValue);
+          setVoteChange(changeValue);
+          setErr(null);
+        })
+        .catch((error) => {
+          setNewVotes(newVotes);
+          setVoteChange(0);
+          setErr("Something went wrong, please try again.");
+        });
+    } else if (voteChange === changeValue) {
+      patchArticle(newArticle.article_id, -changeValue)
+        .then(() => {
+          setNewVotes(newVotes - changeValue);
+          setVoteChange(0);
+          setErr("Something went wrong, please try again.");
```

Comment:
This line here and on line 21 is causing this message to happen when you undo a vote - though it seems to be working fine!